ixmp/backend/jdbc.py

```diff
@@ -875,15 +881,32 @@ def item_delete_elements(self, s, type, name, keys):
         args = (s,) if type == 'set' else (s, type, name)
         self.cache_invalidate(*args)
 
-    def get_meta(self, s):
-        def unwrap(v):
-            """Unwrap meta numeric value (BigDecimal -> Double)"""
-            return v.doubleValue() if isinstance(v, java.BigDecimal) else v
-
-        return {entry.getKey(): unwrap(entry.getValue())
+    def get_meta(self, model: str = None, scenario: str = None,
+                 version: int = None) -> dict:
+
+        meta = self.jobj.getMeta(model, scenario, version)
+        return {entry.getKey(): _unwrap(entry.getValue())
+                for entry in meta.entrySet()}
+
+    def set_meta(self, meta: dict, model: str = None, scenario: str = None,
+                 version = None):
+        if not (model or scenario or version):
+            msg = ('At least one parameter has to be provided out of: '
+                   'model, scenario, version')
```

Comment:
I don't think passing just version makes much sense. It should be one of:
- model
- scenario
- model + scenario
- model + scenario + version